File: gmsh2rea/export_rea.py
def export_rea(filename, elements, boundaries):
    with open(filename, 'w') as file:
        n = elements.shape[0]
        # Add header
        with open('header.rea', 'r') as header_file:
            for line in header_file:
                file.write(line)

        # Mesh
        file.write('**MESH DATA** 6 lines are X,Y,Z;X,Y,Z. Columns corners 1-4;5-8\n')
        file.write(f'{n:8d}   3   {n:8d} NEL NDIM NEL\n')
        for i in range(n):
            file.write(f'            ELEMENT{i+1:12d} [    1A]    GROUP     0\n')
            for j in range(3):
                file.write(f'{elements[i, 0, j]:15.7g}  {elements[i, 1, j]:15.7g}  '
                           f'{elements[i, 2, j]:15.7g}  {elements[i, 3, j]:15.7g}\n')
            for j in range(3):
                file.write(f'{elements[i, 4, j]:15.7g}  {elements[i, 5, j]:15.7g}  '
                           f'{elements[i, 6, j]:15.7g}  {elements[i, 7, j]:15.7g}\n')

        file.write('  ***** CURVED SIDE DATA *****\n')
        file.write('     0 Curved sides follow IEDGE,IEL,CURVE(I),I=1,5, CCURVE\n')
        file.write('  ***** BOUNDARY CONDITIONS ***** \n')
        file.write(' ***** FLUID   BOUNDARY CONDITIONS ***** \n')
        for i in range(n):
            for j in range(6):
                k = [item[2] for item in boundaries if item[0]==i and item[1] == j]
                if len(k) == 0:
                    k = 0
                else:
                    k = k[0]

                if k == 4:
                    file.write(f' W    {i+1:3d} {0:13.5e} {0:13.5e} {0:13.5e} {0:13.5e} {0:13.5e}\n')
                elif k == 2:
                    file.write(f' v    {i+1:3d} {0:13.5e} {0:13.5e} {0:13.5e} {0:13.5e} {0:13.5e}\n')
                elif k == 3:
                    file.write(f' O    {i+1:3d} {0:13.5e} {0:13.5e} {0:13.5e} {0:13.5e} {0:13.5e}\n')
                else:
                    file.write(f' E    {i+1:3d} {0:13.5e} {0:13.5e} {0:13.5e} {0:13.5e} {0:13.5e}\n')

        # Add footer
        with open('footer.rea', 'r') as footer_file:
            for line in footer_file:
                file.write(line)

File: turbineBlade/exportREA.m
function exportREA(filename, elements, boundaries)
  fileID = fopen(filename, 'w');

% Add header
  headerID = fopen('header.rea', 'r');
  while ~feof(headerID)
      line = fgets(headerID);
      fprintf(fileID, '%s', line);
  end
  fclose(headerID);

% Mesh
  fprintf(fileID,'**MESH DATA** 6 lines are X,Y,Z;X,Y,Z. Columns corners 1-4;5-8\n');
  fprintf(fileID,'%8i   %8i   %8i NEL NDIM NEL\n',size(elements,1),3,size(elements,1));
  for i = 1:size(elements,1)
      fprintf(fileID,'            ELEMENT%12i [    1A]    GROUP     0\n',i);
      fprintf(fileID,'%15.7g  %15.7g  %15.7g  %15.7g\n',elements(i,1:4,1));
      fprintf(fileID,'%15.7g  %15.7g  %15.7g  %15.7g\n',elements(i,1:4,2));
      fprintf(fileID,'%15.7g  %15.7g  %15.7g  %15.7g\n',elements(i,1:4,3));
      fprintf(fileID,'%15.7g  %15.7g  %15.7g  %15.7g\n',elements(i,5:8,1));
      fprintf(fileID,'%15.7g  %15.7g  %15.7g  %15.7g\n',elements(i,5:8,2));
      fprintf(fileID,'%15.7g  %15.7g  %15.7g  %15.7g\n',elements(i,5:8,3));
  end

  fprintf(fileID,'  ***** CURVED SIDE DATA *****\n');
  fprintf(fileID,'     0 Curved sides follow IEDGE,IEL,CURVE(I),I=1,5, CCURVE\n');
  fprintf(fileID,'  ***** BOUNDARY CONDITIONS ***** \n');
  fprintf(fileID,' ***** FLUID   BOUNDARY CONDITIONS ***** \n');
  for i = 1:size(elements,1)
    for j = 1:6
        condition = (boundaries(:, 1) == i) & (boundaries(:, 2) == j);
        tag = boundaries(condition, 3);
        if ~isempty(tag)
          tag = tag(1);
          if tag == 1
            fprintf(fileID,' W    %3d %13.5e %13.5e %13.5e %13.5e %13.5e\n', i, 0, 0, 0, 0, 0);
          elseif tag == 2
            fprintf(fileID,' v    %3d %13.5e %13.5e %13.5e %13.5e %13.5e\n', i, 0, 0, 0, 0, 0);
            % fprintf(fileID,' int  %3d %13.5e %13.5e %13.5e %13.5e %13.5e\n', i, 0, 0, 0, 0, 0);
            % fprintf(fileID,' O    %3d %13.5e %13.5e %13.5e %13.5e %13.5e\n', i, 0, 0, 0, 0, 0);
          elseif tag == 3
            fprintf(fileID,' int  %3d %13.5e %13.5e %13.5e %13.5e %13.5e\n', i, 0, 0, 0, 0, 0);
            % fprintf(fileID,' int  %3d %13.5e %13.5e %13.5e %13.5e %13.5e\n', i, 0, 0, 0, 0, 0);
            % fprintf(fileID,' O    %3d %13.5e %13.5e %13.5e %13.5e %13.5e\n', i, 0, 0, 0, 0, 0);
          elseif tag == 4
            fprintf(fileID,' O    %3d %13.5e %13.5e %13.5e %13.5e %13.5e\n', i, 0, 0, 0, 0, 0);
            % fprintf(fileID,' int  %3d %13.5e %13.5e %13.5e %13.5e %13.5e\n', i, 0, 0, 0, 0, 0);
            % fprintf(fileID,' O    %3d %13.5e %13.5e %13.5e %13.5e %13.5e\n', i, 0, 0, 0, 0, 0);
          elseif tag == 5
            fprintf(fileID,' s    %3d %13.5e %13.5e %13.5e %13.5e %13.5e\n', i, 0, 0, 0, 0, 0);
            % fprintf(fileID,' int  %3d %13.5e %13.5e %13.5e %13.5e %13.5e\n', i, 0, 0, 0, 0, 0);
            % fprintf(fileID,' O    %3d %13.5e %13.5e %13.5e %13.5e %13.5e\n', i, 0, 0, 0, 0, 0);
          else
            tag
          end
        else
            fprintf(fileID,' E    %3d %13.5e %13.5e %13.5e %13.5e %13.5e\n', i, 0, 0, 0, 0, 0);
        end
    end
  end
  %
  % fprintf(fileID,'  ***** THERMAL BOUNDARY CONDITIONS *****\n');
  % for i = 1:size(boundaries,1)
  %   for j = 1:6
  %     if boundaries(i,j, 1) == 1
  %       fprintf(fileID,' t    %3d %13.5e %13.5e %13.5e %13.5e %13.5e\n', i, 0, 0, 0, 0, 0);
  %     elseif boundaries(i,j, 1) == 2
  %       fprintf(fileID,' t    %3d %13.5e %13.5e %13.5e %13.5e %13.5e\n', i, 0, 0, 0, 0, 0);
  %     elseif boundaries(i,j, 1) == 3
  %       fprintf(fileID,' t    %3d %13.5e %13.5e %13.5e %13.5e %13.5e\n', i, 0, 0, 0, 0, 0);
  %     else
  %       fprintf(fileID,' E    %3d %13.5e %13.5e %13.5e %13.5e %13.5e\n', i, 0, 0, 0, 0, 0);
  %     end
  %   end
  % end
% Add footer
  footerID = fopen('footer.rea', 'r');
  while ~feof(footerID)
      line = fgets(footerID);
      fprintf(fileID, '%s', line);
  end
  fclose(footerID);
  fprintf(fileID, '\n');

  fclose(fileID);
end
